parser/src/main/java/com/linkedin/avroutil1/parser/avsc/AvscFileParseContext.java

```diff
@@ -178,6 +178,39 @@ public void resolveReferences() {
         for (AvroSchema schema : definedSchemas) {
             resolveReferences(schema);
         }
+
+        // Once all schemas are resolved for this file, we can process the defaults for fields
+        for (AvroSchema schema : definedSchemas) {
+            processAllDefaults(schema);
+        }
+    }
+
+    private void processAllDefaults(AvroSchema schema) {
+        AvroType type = schema.type();
+        if (type == AvroType.RECORD) {
+            AvroRecordSchema recordSchema = (AvroRecordSchema) schema;
+            List<AvroSchemaField> fields = recordSchema.getFields();
+            for (AvroSchemaField field : fields) {
+                SchemaOrRef fieldSchema = field.getSchemaOrRef();
+                boolean hasDefault = field.hasDefaultValue();
+                boolean wasDefinedBefore = fieldSchema.isFullyDefined();
+                if (!wasDefinedBefore) {
+                    if (!fieldSchema.isFullyDefined()) {
+                        if (hasDefault) {
+                            fieldsWithUnparsedDefaults.add(field);
+                        }
+                        continue;
+                    }
+                }
+                //fully defined if we're here
+                if (!hasDefault) {
+                    continue;
+                }
+                if (field.getDefaultValue() instanceof AvscUnparsedLiteral) {
+                    AvscParseUtil.lateParseFieldDefault(field, this);
+                }
+            }
```

Comment:
I think we can refactor these code to remove these two **continue** to make the logic more clear. 

The logic is like:
for (AvroSchemaField field : fields) {
   SchemaOrRef fieldSchema = field.getSchemaOrRef();
   boolean hasDefault = field.hasDefaultValue();
   boolean wasDefinedBefore = fieldSchema.isFullyDefined();  
   if (!wasDefinedBefore) {
       if (hasDefault) {
           fieldsWithUnparsedDefaults.add(field);
       }
   } else {
       if (!hasDefault) {
                if (field.getDefaultValue() instanceof AvscUnparsedLiteral) {
                    AvscParseUtil.lateParseFieldDefault(field, this);
                }
       }
   }
}


Or you can move the the check **if (hasDefault)** ahead:

for (AvroSchemaField field : fields) {
   if (!field.hasDefaultValue()) {
      continue;
   }

   SchemaOrRef fieldSchema = field.getSchemaOrRef();
   boolean wasDefinedBefore = fieldSchema.isFullyDefined();  
   if (!wasDefinedBefore) {
           fieldsWithUnparsedDefaults.add(field);
   } else if (field.getDefaultValue() instanceof AvscUnparsedLiteral) {
           AvscParseUtil.lateParseFieldDefault(field, this);
   }
}